templates/solaris_install.sh.erb

```diff
@@ -1,6 +1,8 @@
 function start_service() {
   service="${1:?}"
-  /opt/puppetlabs/bin/puppet resource service "${service:?}" ensure=running enable=true
+if /opt/puppetlabs/bin/puppet resource service "${service:?}" | grep "ensure => 'running'" ; then
```

Comment:
I'm not sure how this change accomplishes the described outcome.

If the service is already running, then ensure=running does nothing. If it was not running, then how do you know it was running before? Isn't that something we should check before the upgrade?

Also @ekinanp what was here before doesn't make sense to me; the comment suggests we should be doing a service restart but the command here will only start the service if it was stopped.

Reply:
Solaris upgrades are weird, I think we stopped the service prior to doing the upgrade. The service should be restarted in https://github.com/puppetlabs/vanagon/blob/master/resources/solaris/10/postinstall.erb#L39-L42 -- think this is just a sanity check.

I don't remember why we did this, I just remember it being a part of https://github.com/puppetlabs/puppetlabs-puppet_agent/blob/1.5.0/templates/solaris_install.sh.epp prior to my refactor.

Reply:
Found where I'm responsible for originally writing this :)

My suggestion is move the test before "Remove old package", then start services based on that test.

Reply:
@ekinanp thanks for explaining, it's exactly what's intended to do, a sanity check. 👍 
@MikaelSmith I hope I understood this correctly, please have a look again.